spec/controllers/api/v1/offers_controller_spec.rb

```diff
@@ -480,6 +480,23 @@
     before(:each) { generate_and_set_token(reviewer) }
     subject { JSON.parse(response.body) }
 
+    context "Shareable Offers/Published offers filter" do
+      let!(:shareable1) {create :shareable, resource: reviewing_offer}
+      let!(:shareable2) {create :shareable, resource: submitted_offer}
+
+      it "returns offers published on charity site" do
+        get :search, params: { shareable: true }
+        expect(response.status).to eq(200)
+        expect(subject['offers'].size).to eq(2)
+      end
+
+      it "returns offers published on charity site of particular state" do
+        get :search, params: { shareable: true, state: "submitted" }
+        expect(response.status).to eq(200)
+        expect(subject['offers'].size).to eq(1)
```

Comment:
Along with checking count, you can also verify if the correct offer is returned in the response or not.

Reply:
sure..